src/Scanner.js

```diff
@@ -279,6 +369,117 @@ class DefaultScanner {
         return input;
     }
 
+    /**
+     * Alternative implementation using closest() API to find form-like containers
+     * @param {HTMLElement|HTMLInputElement|HTMLSelectElement} input
+     * @returns {HTMLFormElement|HTMLElement}
+     */
+    getParentFormV2(input) {
+        // First check for native form element
+        if ((input instanceof HTMLInputElement || input instanceof HTMLSelectElement) && input.form) {
+            // Use input.form unless it encloses most of the DOM
+            if (!isFormLikelyToBeUsedAsPageWrapper(input.form)) {
+                return input.form;
+            }
+        }
+
+        // Try using closest() with form-like selectors
+        const formSelectors = [
+            // Explicit forms first
+            'form:not([role=search])',
```

Comment:
If the input has a parent form, you'd get it with `input.form`, so this is redundant.